eppo-sdk-test/EppoClientTest.cs

```diff
@@ -65,15 +81,96 @@ private static string GetMockFlagConfig()
     }
 
     [Test]
-    public void ShouldLogAssignment()
+    public void ShouldReturnDefaultInUnrefreshedClientMode()
+    {
+        var client = CreteClientModeClient();
+        var alice = new Dictionary<string, object>()
+        {
+            ["email"] = "alice@company.com",
+            ["country"] = "US"
+        };
+        var result = client.GetIntegerAssignment("integer-flag", "alice", alice, 1);
+
+        // Assert result is the default value of `1`.
+        That(result, Is.EqualTo(1));
+
+        _mockServer!.Should()
+            .HaveReceivedNoCalls();
+    }
+
+    [Test]
+    public void ShouldReturnAssignmentsAfterRefreshInClientMode() {
+                var client = CreteClientModeClient();
+        var alice = new Dictionary<string, object>()
+        {
+            ["email"] = "alice@company.com",
+            ["country"] = "US"
+        };
+
+        var noConfigResult = client.GetIntegerAssignment("integer-flag", "alice", alice, 1);
+
+        // Assert result is the default value of `1`.
+        That(noConfigResult, Is.EqualTo(1));
+
+        client.RefreshConfiguration();
+
+        var loadedResult  = client.GetIntegerAssignment("integer-flag", "alice", alice, 1);
+        That(loadedResult, Is.EqualTo(3));
+    }
+
+
+    [Test]
+    public void ShouldRunInClientMode()
```

Comment:
Do we need to mock time here and speed up time and make sure no more polling calls made?

Reply:
mocking time and rewriting this dependency are more effort than it's worth. so we poll on a very short interval and have a short sleep